pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -192,6 +209,27 @@ def _add_prefix(
         count = seen_optimizer_types[optimizer_cls]
         return name + f'-{count - 1}' if count > 1 else name
 
+    def _add_suffix(self, name: str, param_groups: List[Dict], param_group_index: int, use_names: bool = True) -> str:
+        if len(param_groups) > 1:
+            if not use_names:
+                return f'{name}/pg{param_group_index+1}'
+            else:
+                pg_name = param_groups[param_group_index].get('name', f'pg{param_group_index+1}')
+                return f'{name}/{pg_name}'
+        elif use_names:
+            pg_name = param_groups[param_group_index].get('name')
+            return f'{name}/{pg_name}' if pg_name else name
+        else:
+            return name
+
+    def _duplicate_param_group_names(self, param_groups: List[Dict]) -> Set[str]:
+        names = [pg.get('name', f'pg{i+1}') for i, pg in enumerate(param_groups)]
+        unique = set(names)
+        if len(names) == len(unique):
+            return set()
+        else:
+            return set(n for n in names if names.count(n) > 1)
```

Comment:
```suggestion
        return set(n for n in names if names.count(n) > 1)
```

Reply:
I explicitly avoided this pattern for perf. implications. Ideally, the `_duplicate_param_group_names` should be a validation check _only_ for the happy path, which is satisfied with the following check:

```python
 unique = set(names)        
 if len(names) == len(unique):
     return set()
 ```

This should be very fast as it will (optionally) get called every `on_train_batch_start`. The failure case will ultimately raise an exception, but I wanted the exception to provide actionable info on what ultimately caused the issue "the duplicate param-group names" in this case.

```python
else:
    return set(n for n in names if names.count(n) &gt; 1)
```

For this, I chose to trade off performance for readability as, mentioned above, an exception will ultimately be raised in this case, disallowing repeated calls.

In sum, I prefer the original implementation; but it's your codebase :)

Reply:
I am also open to being wrong because I missed something 😅

Reply:
Haha, good enough reason! Go for it for the changes! 😄

Reply:
I do appreciate the suggestion! Your suggestion def. lowers the surface area, which is an understandable priority for plugged-in maintainers!!